vspace/src/main/java/edu/asu/diging/vspace/core/file/impl/StorageEngine.java

```diff
@@ -130,4 +138,101 @@ public boolean renameImage(IVSImage image, String newFileName) {
+     * 
+     * @param folderName - name of the folder to be zipped
+     * @return byte[] - zipped data as a byte array
+     * @throws IOException if an I/O error occurs while reading files or writing the zip data
+     */
+    @Override
+    public byte[] generateZip(String folderName) throws IOException {
+        Path zipFile = Paths.get(path + File.separator + folderName);
+        ByteArrayOutputStream  byteArrayOutputStream = new ByteArrayOutputStream();
+        Path zipFilePath = Paths.get(File.separator + folderName+".zip");
+        FileOutputStream fileOutputStream = new FileOutputStream(getFile("",folderName+".zip"));
+        try (BufferedOutputStream bufferedOutputStream = new BufferedOutputStream(fileOutputStream);
+                ZipOutputStream responseZipStream = new ZipOutputStream(bufferedOutputStream);
+                Stream<Path> paths = Files.walk(zipFile)) {
+            paths
+                .filter(path -> !Files.isDirectory(path))
+                .forEach(path -> {
+                    ZipEntry  zipEntry = new ZipEntry(zipFile.relativize(path).toString());
+                    try {
+                        responseZipStream.putNextEntry(zipEntry);
+                        Files.copy(path, responseZipStream);
+                        responseZipStream.closeEntry();
+                        //delete the folder 
+                        Files.delete(zipFile);
+                    } catch (IOException e) {
+                        logger.error("Could not generate Zip folder", e);
+                    }
+                });            
+        }
+        return byteArrayOutputStream.toByteArray();        
+    }
+    
+    /**
+     * Copies the contents of a specified folder to a destination folder in the relative path.
+     *
+     * @param relativePath - the relative path within which the folder contents will be copied
+     * @param folderToCopy - the path of the folder whose contents are to be copied
+     * @throws IOException - if an I/O error occurs during the copy process
+     */
+    @Override
+    public void copyToFolder(String relativePath, String folderToCopy) throws IOException {
+        FileUtils.copyDirectory(new File(folderToCopy), new File(path + File.separator+ relativePath));
+    }
+    
+    /**
+     * Deletes the specified folder
+     *
+     *@param folderPath  path to the folder to be deleted
+     */
+    @Override
+    public void deleteFolder(String folderPath) throws IOException {
+        FileUtils.deleteDirectory(getFile("",folderPath));
+    }
+    
+    /**
+     * To create a zip of a given folder
+     * 
+     * @param zipFilename - name of the folder to be zipped
+     * @return byte[] - zipped data as a byte array
+     * @throws ExhibitionSnapshotNotFoundException 
+     * @throws IOException 
+     */
+    @Override
+    public byte[] downloadZip(String zipFilename) throws ExhibitionSnapshotNotFoundException, IOException, FileNotFoundException {
```

Comment:
`downloadZip` seems like the wrong name if the method zips a folder.

Reply:
actually, the javadoc seems wrong; also `download` is the wrong word, should probably be `getZip` or something.